### PR TITLE
Enhancements to e2e test, adding Autoscaling

### PR DESCRIPTION
**Pull request by blublinsky**
## Why are these changes needed?

Cleaning up existing e2e tests
Adding e2e autoscaling test

## Checks

- [x ] I've made sure the tests are passing. 
- Testing Strategy
   - [ x] Unit tests
   - [x] e2e tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by blublinsky** [comment]
@z103cb , @tedhtchang

cc @kevin85421

**Review by z103cb — COMMENTED** [review]
Other than small nitpicks around comments in code this is PR is LGTM.

File: apiserver/test/e2e/cluster_server_autoscaler_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	actualJob, actualRpcStatus, err = tCtx.GetRayApiServerClient().CreateRayJob(&delete_actor_request)
+	require.NoError(t, err, "No error expected")
+	require.Nil(t, actualRpcStatus, "No RPC status expected")
+	require.NotNil(t, actualJob, "A job is expected")
+	waitForRayJob(t, tCtx, create_actor_request.Job.Name, rayv1api.JobStatusSucceeded)
+
+	// worker pod should be deleted in 30 sec. In reality its larger
+	time.Sleep(100 * time.Second)
+
+	// Get number of workers
```
[the remark this answers is not in the document]
Comment (blublinsky) [reply apiserver/test/e2e/cluster_server_autoscaler_e2e_test.go:152]: Fixed

File: apiserver/test/e2e/cluster_server_autoscaler_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	}
+
+	// Create cluster
+	actualCluster, actualRpcStatus, err := tCtx.GetRayApiServerClient().CreateCluster(&cluster_req)
+	require.NoError(t, err, "No error expected")
+	require.Nil(t, actualRpcStatus, "No RPC status expected")
+	require.NotNil(t, actualCluster, "A cluster is expected")
+	waitForRunningCluster(t, tCtx, actualCluster.Name)
+
+	// Get number of workers
```
[the remark this answers is not in the document]
Comment (blublinsky) [reply apiserver/test/e2e/cluster_server_autoscaler_e2e_test.go:97]: Yes, they are in the status

File: apiserver/test/e2e/cluster_server_e2e_test.go [hunk 3]
```diff
@@ -537,8 +537,16 @@ func TestGetAllClusters(t *testing.T) {
 	require.Nil(t, actualRpcStatus, "No RPC status expected")
 	require.NotNil(t, response, "A response is expected")
 	require.NotEmpty(t, response.Clusters, "A list of clusters is required")
-	require.Equal(t, tCtx.GetRayClusterName(), response.Clusters[0].Name)
-	require.Equal(t, tCtx.GetNamespaceName(), response.Clusters[0].Namespace)
+	got_cluster := false
```
[the remark this answers is not in the document]
Comment (blublinsky) [reply apiserver/test/e2e/cluster_server_e2e_test.go:540]: Fixed

File: apiserver/test/e2e/cluster_server_e2e_test.go [hunk 4]
```diff
@@ -567,8 +575,16 @@ func TestGetClustersInNamespace(t *testing.T) {
 	require.Nil(t, actualRpcStatus, "No RPC status expected")
 	require.NotNil(t, response, "A response is expected")
 	require.NotEmpty(t, response.Clusters, "A list of compute templates is required")
-	require.Equal(t, tCtx.GetRayClusterName(), response.Clusters[0].Name)
-	require.Equal(t, tCtx.GetNamespaceName(), response.Clusters[0].Namespace)
+	got_cluster := false
```
[the remark this answers is not in the document]
Comment (blublinsky) [reply apiserver/test/e2e/cluster_server_e2e_test.go:578]: Fixed

**Comment by blublinsky** [comment]
@kevin85421 its all yours, ready to commit

**Review by kevin85421 — APPROVED** [review]
This PR is only for KubeRay API server. Stamp.

**Pull request merged**